File: components/permissions/src/main/java/org/cloudfoundry/credhub/audit/entities/GetPermissions.java
package org.cloudfoundry.credhub.audit.entities;

import org.cloudfoundry.credhub.audit.OperationDeviceAction;
import org.cloudfoundry.credhub.audit.RequestDetails;

public class GetPermissions implements RequestDetails {

  private String credentialName;

  public GetPermissions() {
    super();

  }

  public GetPermissions(final String credentialName) {
    super();
    this.credentialName = credentialName;
  }

  public String getCredentialName() {
    return credentialName;
  }

  public void setCredentialName(final String credentialName) {
    this.credentialName = credentialName;
  }

  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.GET_PERMISSIONS;
  }
}

File: components/permissions/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/GetPermissions.kt
package org.cloudfoundry.credhub.audit.entities

import org.cloudfoundry.credhub.audit.OperationDeviceAction
import org.cloudfoundry.credhub.audit.RequestDetails

class GetPermissions(
    credentialName: String,
) : RequestDetails {
    override fun operation(): OperationDeviceAction = OperationDeviceAction.GET_PERMISSIONS
}
